docs/spark_processor.md

```diff
@@ -0,0 +1,21 @@
+# Spark Processor
+
+The SparkProcessor does feature ETL using Spark as the processing engine. In the
+following sections we describe the deployment modes supported by SparkProcessor
+and the configuration keys accepted by each mode.
+
+## Deployment Mode
+FeatHub's SparkProcessor only supports Spark's local mode. When running in local
```

Comment:
Can you provide link to the Spark official doc that defines the `local` mode?

I could not find local mode. The `spark-submit` accepts argument `--deploy-mode` with values such as `client` and `cluster`.

Reply:
Spark's document has not provided a definition of local mode. The closest document might be this[1], having been added to the "Configurations" section.

Spark's deployment mode has two dimensions. The first dimension is local/standalone/yarn/k8s, and the second is client/cluster. `--deployment-mode` configures the second dimension, and `--master` configures the first. Similar dimensions also exist in Flink, where we configure Standalone/Yarn/K8S and Application/Session/Per-job mode.

[1] https://spark.apache.org/docs/3.3.1/submitting-applications.html#master-urls